#1497 support analyze ET in MLSQL

hellozepp opened the pull request:
#1497

Commit:
Add an ET plugin SyntaxAnalyzeExt, the fix is as follows:
1. Add function modelType, doc, codeExample; implement trait ETAuth
2. Fix ET output problem. before: Only log the tables. after: return specific tables to DF

allwefantasy commented on external/mlsql-ets/src/main/java/tech/mlsql/plugins/ets/SyntaxAnalyzeExt.scala:
```diff
@@ -0,0 +1,40 @@
+/**
+  * 09/08/2021 hellozepp(lisheng.zhanglin@163.com)
+  */
+class SyntaxAnalyzeExt(override val uid: String) extends SQLAlg with WowParams with Logging with WowLog {
```
Please overwrite functions e.g.  modelType, doc, codeExample; implement trait `ETAuth`；  Take class ` streaming.dsl.mmlib.algs.SQLRandomForest` as example.

hellozepp replied:
I will fix it.

allwefantasy commented on external/mlsql-ets/src/main/java/tech/mlsql/plugins/ets/SyntaxAnalyzeExt.scala:
```diff
@@ -0,0 +1,40 @@
+    val context = ScriptSQLExec.contextGetOrForTest()
+    val _sql = params(sql.name)
+    val identifiers = MLSQLAuthParser.filterTables(_sql, context.execListener.sparkSession)
+    logInfo(format(identifiers.map(_.table).toList.mkString("[",",","]")))
```
So we just log the tables we extract from sql?  How can we get the result ?

hellozepp replied:
Need to be converted to DF, I will fix it

allwefantasy commented on external/mlsql-ets/src/main/java/tech/mlsql/plugins/ets/SyntaxAnalyzeExt.scala:
```diff
@@ -23,11 +44,12 @@ class SyntaxAnalyzeExt(override val uid: String) extends SQLAlg with WowParams w
-        logInfo(format("table list :" + identifiers.map(_.table).toList.mkString("[", ",", "]")))
-      case _ =>
+        val tables = identifiers.map(_.table)
+        logInfo(format("table list :" + tables.toList.mkString("[", ",", "]")))
```
This log is useless.

hellozepp replied:
Ok i will delete it

allwefantasy commented on external/mlsql-ets/src/main/java/tech/mlsql/plugins/ets/SyntaxAnalyzeExt.scala:
```diff
@@ -41,5 +63,47 @@ class SyntaxAnalyzeExt(override val uid: String) extends SQLAlg with WowParams w
+      |select * from (select * from table1 as c) as d left join table2 as e on d.id=e.id
+      |''';
+      |
+      |!lastCommand named table3;
```
No need to use` !lastCommand`。 Try this 

```
run command as SyntaxAnalyzeExt. `` where
action = "extractTables" and sql=''' ... '''
as extractedTables;

select * from extractedTables as output;
```

hellozepp replied:
got it

allwefantasy commented on external/mlsql-ets/src/main/java/tech/mlsql/plugins/ets/SyntaxAnalyzeExt.scala:
````diff
@@ -0,0 +1,105 @@
+      |```
+      |Note that the value of the parameter `sql` should not be added with a semicolon.""".stripMargin)
+
+  override def auth(etMethod: ETMethod, path: String, params: Map[String, String]): List[TableAuthResult] = List()
````
Auth should not be empty.

allwefantasy commented on external/mlsql-ets/src/main/java/tech/mlsql/plugins/ets/SyntaxAnalyzeExt.scala:
```diff
@@ -0,0 +1,105 @@
+      case "extractTables" =>
+        import df.sparkSession.implicits._
+        MLSQLAuthParser.filterTables(_sql, context.execListener.sparkSession).map(_.table).toDF("tableName")
+      case _ => throw new NoSuchElementException("Failed to execute SyntaxAnalyzeExt, unsupported statement")
```
unsupported statement?  maybe `unsupported action` more clear? Also ,since you have only one action for now, can we  set action `extractTables` as default.

hellozepp replied:
Yes it will be better

Commit:
Add an ET plugin SyntaxAnalyzeExt, the fix is as follows:
1. Add function modelType, doc, codeExample; implement trait ETAuth
2. Fix ET output problem. before: Only log the tables. after: return specific tables to DF
3. Add Param docs
